frontend/src/AppRoot.tsx

```diff
@@ -1,28 +1,54 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import { useAuth0 } from '@auth0/auth0-react'
 import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
 import { ApolloProvider } from '@apollo/client'
-import apolloClientInstance from './data/apolloClientInstance'
+
+import makeApolloClient from './data/apolloClientInstance'
 import ApolloDemoPage from './pages/ApolloDemo'
 import HomePage from './pages/Home'
 import PublicHomePage from './pages/PublicHome'
 import GroupList from './pages/groups/GroupList'
 import PrivateRoute from './components/PrivateRoute'
 
+const fetchProfile = (token: string) => {
+  return fetch('/profile', {
+    headers: { Authorization: `Bearer ${token}` },
+  })
+}
```

Comment:
Thought: I wish we didn't have to pass the auth token manually here. I would also prefer to use GraphQL for all the requests on the frontend instead of writing custom endpoints on the server. I'll mess with this to see if that's possible.

Reply:
Part of the reason I made it separate from GraphQL was to keep things simple in the server-side GraphQL context provider -- this way we can assert that there's a user account for all GraphQL requests or throw.

Reply:
Gotcha. I don't know if this is possible with GraphQL, but it'd be sweet if we could intercept the query below and run some custom code on the server:

```js
const { data } = useQuery(gql`
  query GetCurrentUser {
    user($userId: string) {
      id
      name
    }
  }
`

Reply:
I feel like it's one of those things we'll iterate on as we learn about the tech though.